api/loader/fileloader.go

```diff
@@ -180,8 +180,17 @@ func (fl *fileLoader) New(path string) (ifc.Loader, error) {
 	}
 
 	if filepath.IsAbs(path) {
-		return nil, fmt.Errorf("new root '%s' cannot be absolute", path)
+		if !strings.HasPrefix(path, fl.root.String()) {
+			return nil, fmt.Errorf("referenced absolute path '%s' escapes target directory", path)
+		}
+
+		// Convert absolute path to relative path for unified handling
+		path, err = filepath.Rel(fl.root.String(), path)
```

Comment:
Have you tried to test this? Tried it quickly and it is not converting the absolute path to relative one.

Reply:
I had forgotten to push the other half of the commit to fix this part on the `lcloader` as well. It should work with the newly pushed content.